tests/engine/test_state.py

```diff
@@ -749,3 +752,14 @@ def __init__(self, *args) -> None:
 
     # Does not alter the original object
     assert isinstance(state.result, UnpicklableException)
+
+
+@patch("prefect.engine.state.super")
+def test_state_sizeof(mock_super):
+    state = State(result=10)
+    mock_super().__sizeof__.return_value = 55
+    # Don't use sys.getsizeof(state) here because
+    # getsizeof() includes overhead that is hard
+    # to account for.
+    # Instead call __sizeof__ directly
+    assert state.__sizeof__() == 55 + sys.getsizeof(state.result)
```

Comment:
This LGTM but I'll have @abrookins take a look too.

Perhaps one additional assertion that `sys.getsizeof(state) >= state.__sizeof__()` so we can be sure it's being applied in practice?

Reply:
Sure thing, I can add that to the test

Reply:
I declare retroactively that this looks good to me. 😄